samples/node/text_generation.js

```diff
@@ -0,0 +1,258 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GoogleGenerativeAI } from "@google/generative-ai";
+
+async function textGenTextOnlyPrompt() {
+  // [START text_gen_text_only_prompt]
+  const genAI = new GoogleGenerativeAI(process.env.API_KEY);
+  const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });
+
+  const prompt = "Write a story about a magic backpack.";
+
+  const result = await model.generateContent(prompt);
+  const response = result.response;
+  const text = response.text();
+  console.log(text);
+  // [END text_gen_text_only_prompt]
+}
+
+async function textGenTextOnlyPromptStreaming() {
+  // [START text_gen_text_only_prompt_streaming]
+  const genAI = new GoogleGenerativeAI(process.env.API_KEY);
+  const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });
+
+  const prompt = "Write a story about a magic backpack.";
+
+  const result = await model.generateContentStream(prompt);
+
+  // print text as it comes in
+  for await (const chunk of result.stream) {
+    const chunkText = chunk.text();
+    console.log(chunkText);
+  }
+  // [END text_gen_text_only_prompt_streaming]
+}
+
+async function textGenMultimodalOneImagePrompt() {
+  // [START text_gen_multimodal_one_image_prompt]
+  const genAI = new GoogleGenerativeAI(process.env.API_KEY);
+  const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });
+
+  function fileToGenerativePart(path, mimeType) {
+    return {
+      inlineData: {
+        data: Buffer.from(fs.readFileSync(path)).toString("base64"),
+        mimeType,
+      },
+    };
+  }
+
+  const prompt = "Describe how this product might be manufactured";
+  // Note: The only accepted mime types are some image types, image/*.
+  const imagePart = fileToGenerativePart("./utils/jetpack.jpg", "image/jpeg");
+
+  const result = await model.generateContent([prompt, imagePart]);
+  const response = result.response;
+  const text = response.text();
+  console.log(text);
+  // [END text_gen_multimodal_one_image_prompt]
+}
+
+async function textGenMultimodalOneImagePromptStreaming() {
+  // [START text_gen_multimodal_one_image_prompt_streaming]
+  const genAI = new GoogleGenerativeAI(process.env.API_KEY);
+  const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });
+
+  function fileToGenerativePart(path, mimeType) {
+    return {
+      inlineData: {
+        data: Buffer.from(fs.readFileSync(path)).toString("base64"),
+        mimeType,
+      },
+    };
+  }
+
+  const prompt = "Describe how this product might be manufactured";
+  // Note: The only accepted mime types are some image types, image/*.
+  const imagePart = fileToGenerativePart("./utils/jetpack.jpg", "image/jpeg");
+
+  const result = await model.generateContentStream([prompt, imagePart]);
+
+  // print text as it comes in
```

Comment:
Add a period. I think this repeats, so I'll stop commenting.

Reply:
Done.